contracts/pair_stable_bluna/src/contract.rs

```diff
@@ -0,0 +1,1916 @@
+            contract_addr: bluna_rewarder.to_string(),
+            msg: to_binary(&anchor_basset::reward::ExecuteMsg::ClaimRewards {
+                recipient: Some(bluna_reward_holder.to_string()),
+            })?,
+            funds: vec![],
+        }),
+        CosmosMsg::Wasm(WasmMsg::Execute {
+            contract_addr: env.contract.address.to_string(),
+            funds: vec![],
+            msg: to_binary(&ExecuteMsg::HandleReward {
+                previous_reward_balance: reward_balance,
+                user_share,
+                total_share,
+                user: info.sender.clone(),
+                receiver,
+            })?,
+        }),
+    ])
+}
+
+/// ## Description
+/// Claims bluna reward and sends it to the specified receiver
+/// Returns an [`ContractError`] on failure, otherwise returns the [`Response`] with the
+/// specified attributes if the operation was successful.
+/// ## Params
+/// * **deps** is the object of type [`Deps`].
+///
+/// * **env** is the object of type [`Env`].
+///
+/// * **info** is the object of type [`MessageInfo`].
+///
+/// * **receiver** is object of type [`Addr`]
+fn claim_reward(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    receiver: Option<String>,
+) -> Result<Response, ContractError> {
+    let receiver = receiver
+        .map(|receiver| addr_validate_to_lower(deps.api, &receiver))
+        .transpose()?
+        .unwrap_or_else(|| info.sender.clone());
+
+    let config: Config = CONFIG.load(deps.storage)?;
+    let user_share = query_token_balance(
+        &deps.querier,
+        config.pair_info.liquidity_token.clone(),
+        info.sender.clone(),
+    )?;
+
+    let total_share: Uint128 = query_supply(&deps.querier, config.pair_info.liquidity_token)?;
+
+    Ok(
+        Response::new().add_messages(get_bluna_reward_handling_messages(
+            deps.as_ref(),
+            &env,
+            &info,
+            config.bluna_rewarder.as_str(),
+            user_share,
+            total_share,
+            receiver,
+        )?),
+    )
+}
+
+/// ## Description
+/// Handles and distributes rewards.
+/// Returns an [`ContractError`] on failure, otherwise returns the [`Response`] with the
+/// specified attributes if the operation was successful.
+/// ## Params
+/// * **deps** is the object of type [`DepsMut`].
+///
+/// * **env** is the object of type [`Env`].
+///
+/// * **info** is the object of type [`MessageInfo`].
+///
+/// * **previous_reward_balance** is object of type [`Uint128`].
+///
+/// * **user_share** is object of type [`Uint128`].
+///
+/// * **total_share** is object of type [`Uint128`].
+///
+/// * **user** is object of type [`Addr`].
+///
+/// * **receiver** is object of type [`Addr`].
+#[allow(clippy::too_many_arguments)]
+pub fn handle_reward(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    previous_reward_balance: Uint128,
+    user_share: Uint128,
+    total_share: Uint128,
+    user: Addr,
+    receiver: Addr,
```

Comment:
Can we make this field optional, and if it is not specified, then send rewards to the user?

Reply:
It is callback, and info.sender is the contract itself here, not a user who claims reward

Reply:
I'm sorry, we can